stdlib: Enable upcoming `MemberImportVisibility` feature

tshortli opened the pull request:
It was already supposed to be enabled, but I forgot to update the CMake when I renamed the feature from `ExtensionImportVisibility` to `MemberImportVisibility`.

However, leave it disabled on the CxxStdlib module since the C++ standard library's modularization varies a lot by platform, making it difficult to add the right conditional imports.

tshortli committed:
stdlib: Enable upcoming MemberImportVisibility feature.

It was already supposed to be enabled, but I forgot to update the CMake when I
renamed the feature from ExtensionImportVisibility to MemberImportVisibility.

However, leave it disabled on the CxxStdlib module since the C++ standard
library's modularization varies a lot by platform, making it difficult to add
the right conditional imports.

tshortli commented:
@swift-ci please smoke test

tshortli commented:
@swift-ci please smoke test

tshortli commented:
@swift-ci please smoke test

tshortli commented:
Now depends on https://github.com/swiftlang/swift/pull/77662.

tshortli commented:
@swift-ci please smoke test

tshortli commented:
@swift-ci please smoke test

tshortli commented:
@swift-ci please smoke test

tshortli commented:
https://github.com/swiftlang/swift-driver/pull/1730

@swift-ci please smoke test

egorzhdan commented on stdlib/public/Cxx/CMakeLists.txt:
```diff
@@ -25,6 +25,11 @@ add_swift_target_library(swiftCxx STATIC NO_LINK_NAME IS_STDLIB IS_SWIFT_ONLY
     # For functionality that depends on the C++ stdlib, use C++ stdlib overlay (`swiftstd` module).
     -Xcc -nostdinc++
 
+    # The varying modularization of the C++ standard library on different
+    # platforms makes it difficult to enable MemberImportVisibility for this
+    # module
+    -disable-upcoming-feature MemberImportVisibility
```
I think this flag should probably only be on the CxxStdlib module. The Cxx module doesn't import the C++ standard library and is mostly platform-independent.

tshortli committed:
Driver: Pass -disable-upcoming-feature and -disable-experimental-feature to the frontend.

The legacy driver is still used to build the compiler on Windows, so we need it
to handle these new flags correctly in order to adopt them in the stdlib.

tshortli commented:
https://github.com/swiftlang/swift-driver/pull/1730

@swift-ci please smoke test

nkcsgexi commented:
🎉
